projects/extension/sql/idempotent/900-semantic-catalog-init.sql

```diff
@@ -0,0 +1,69 @@
+--FEATURE-FLAG: text_to_sql
+
+-------------------------------------------------------------------------------
+-- initialize_semantic_catalog
+create or replace function ai.initialize_semantic_catalog
+( "name" pg_catalog.name default 'default'
+, embedding pg_catalog.jsonb default null
+, indexing pg_catalog.jsonb default ai.indexing_default()
+, scheduling pg_catalog.jsonb default ai.scheduling_default()
+, processing pg_catalog.jsonb default ai.processing_default()
+, grant_to pg_catalog.name[] default ai.grant_to()
+) returns pg_catalog.int4
+as $func$
+declare
+    _catalog_id pg_catalog.int4;
+    _obj_vec_id pg_catalog.int4;
+    _sql_vec_id pg_catalog.int4;
+begin
+    grant_to = pg_catalog.array_cat
+    ( grant_to
+    , array
+      [ pg_catalog."session_user"()
+      , 'pg_database_owner'::name
+      ]
+    );
+
+    insert into ai.semantic_catalog("name")
+    values (initialize_semantic_catalog."name")
+    returning id
+    into strict _catalog_id
+    ;
+
+    select ai.create_vectorizer
+    ( 'ai.semantic_catalog_obj'::pg_catalog.regclass
+    , destination=>pg_catalog.format('semantic_catalog_obj_%s', _catalog_id)
+    , embedding=>embedding
+    , indexing=>indexing
+    , scheduling=>scheduling
+    , processing=>processing
+    , grant_to=>grant_to
+    , formatting=>ai.formatting_python_template() -- TODO: this ain't gonna work
```

Comment:
What's your plan here?

Reply:
I have two options:

1. implement custom python formatting for pgai and then use it here
2. add another text column to the table and use a trigger to do my formatting, and then use that column for embedding

I haven't decided which to pursue yet

Reply:
part of me is leaning towards trigger because there are certain SQL functions that will help me do the formatting I want to do.

It kinda makes me want to do a custom SQL formatting option.